### PR TITLE
Change deprecated method getDrawable. Fix #522

### LINKED ISSUE #522: Change deprecated method getDrawable
There are several uses of deprecated method getDrawable across the code base leading to compilation warnings that look like this:

     warning: [deprecation] getDrawable(int) in Resources has been deprecated
     marker.setIcon(getResources().getDrawable(R.drawable.ic_place_black_36dp));

change this methods.

### PR DESCRIPTION
**Pull request by vikasdesale**
@lognaturel Please check the PR and let me know if any change required.

Thank you :slightly_smiling_face:

### CODE REVIEW

**Review by lognaturel — CHANGES_REQUESTED** [review]
Using the `ResourcesCompat` method you have used means no theme will be taken into account, even on devices running API >= 21. 

You should instead use the `ContextCompat` [getDrawable method](http://stackoverflow.com/a/29041466/137744). 

Additionally, please make sure you have a space after commas in your argument lists. Thank you!

**Comment by vikasdesale** [comment]
Please review @lognaturel. I have done it. thank you!

File: collect_app/src/main/java/org/odk/collect/android/activities/GeoPointOsmMapActivity.java [hunk 1]
```diff
@@ -28,6 +28,8 @@
 import android.os.Looper;
 import android.provider.Settings;
 import android.support.v4.app.FragmentActivity;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.res.ResourcesCompat;
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/GeoPointOsmMapActivity.java]: No longer needed!

File: collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceOsmMapActivity.java [hunk 2]
```diff
@@ -31,6 +31,8 @@
 import android.os.Looper;
 import android.preference.PreferenceManager;
 import android.provider.Settings;
+import android.support.v4.content.ContextCompat;
+import android.support.v4.content.res.ResourcesCompat;
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceOsmMapActivity.java]: No longer needed!

File: collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceOsmMapActivity.java [hunk 3]
```diff
@@ -434,7 +436,7 @@ public void overlayIntentTrace(String str) {
             marker.setOnMarkerClickListener(nullmarkerlistner);
             marker.setDraggable(true);
             marker.setOnMarkerDragListener(draglistner);
-            marker.setIcon(getResources().getDrawable(R.drawable.ic_place_black_36dp));
+            marker.setIcon(ContextCompat.getDrawable(getApplicationContext(),R.drawable.ic_place_black_36dp));
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceOsmMapActivity.java]: add space after comma

Comment (vikasdesale) [reply collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceOsmMapActivity.java]: Sorry @lognaturel.

**Comment by vikasdesale** [comment]
please review @lognaturel. sorry once again for wasting your time. I will take care of these small issues from next time. 
Thank you!

**Review by lognaturel — APPROVED** [review]

**Comment by lognaturel** [comment]
Thanks, @vikasdesale! Don't worry, the important thing is that we keep learning. 😊

**Pull request merged**